src/token-tracker-link.ts

```diff
@@ -4,10 +4,16 @@ export = function getTokenTrackerLink(
   tokenAddress: string,
   network: string,
   holderAddress?: string,
+  customNetworkUrl?: string,
 ): string {
+  if(customNetworkUrl?.length){
+    let parsedUrl = new URL(`token/${tokenAddress}`,customNetworkUrl)
```

Comment:
Unfortunately it's not quite this easy. The "base" URL passed to the URL constructor as the second param is only used for its origin. Any path or query parameters are discarded.

I'd suggest something like this:

```suggestion
    const { username, password, protocol, host, pathname, search, hash } = new URL(customNetworkUrl)
    const newPath = pathname.endsWith('/') ? `${pathname}token/${tokenAddress}` : `${pathname}/token/${tokenAddress}` 
    const auth = username ? `${username}:${password}` : ''
    let parsedUrl = new URL(`${protocol}//${auth}${host}${newPath}${search}${hash}`)
```

That should preserve all pieces of the URL while adding a path.

Check out this diagram for a reference of all of these different URL parts: https://nodejs.org/api/url.html#url_url_strings_and_url_objects
I found it helpful.

Reply:
Oh and the same advice applies to getCustomBlockExplorerAccountLink and getExplorerLink as well.

Maybe this URL manipulation can be moved to a common function :thinking: Something like `addPathToUrl`

Reply:
Thank you for showing me this 🙂 was trying to work with the URL API and I had a feeling I wasn't using it the way I really wanted to lol. The docs on it weren't the most amazing imo 😅

Reply:
I had the same experience! The MDN docs are _OK_ but it's really hard to tell what it does overall without a nice diagram like this. This one I linked was meant to be a comparison between the Node.js URL API and the standard WHATWG URL API, rather than a reference for either. But even still I always refer back to it because it's so much easier to understand than the docs.

Reply:
After reviewing I don't believe this comment was addressed unless I'm missing something. I think we should add some test cases for URLs that contain paths, auth or query parameters. Those tests can then be grouped to test the recommended utility function `addPathToUrl`

Reply:
Whoops! Thanks for the heads up and gotcha!